pkg/addon/addon.go

```diff
@@ -489,7 +543,32 @@ func RenderApp(addon *Addon, config *rest.Config, args map[string]interface{}) (
 				Name: "deploy-runtime",
 				Type: "deploy2runtime",
 			})
-	} else {
+	case addon.Name == "observability":
+		arg, ok := args["domain"]
```

Comment:
Please create an issue and assign me，we should  use some  mechanism  to refactor these hack code.

Reply:
https://github.com/oam-dev/catalog/issues/193